examples/esp32/xively_demo/main/xively_if.c

```diff
@@ -0,0 +1,617 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/event_groups.h"
+#include "freertos/task.h"
+
+#include "xively_if.h"
+
+/******************************************************************************
+*                          Xively Interface Macros
+******************************************************************************/
+#define DELAY_MS( t ) vTaskDelay( t / portTICK_PERIOD_MS )
+
+#define XIF_BUTTON_TOPIC_NAME "Button"
+#define XIF_LED_TOPIC_NAME "LED"
+
+/******************************************************************************
+*                     Xively Interface Structs and Enums
+******************************************************************************/
+typedef struct
+{
+    char* xi_account_id;
+    char* xi_device_id;
+    char* xi_device_pwd;
+} xif_device_info_t;
+
+typedef enum
+{
+    XIF_MQTT_DISCONNECTED = 0,
+    XIF_MQTT_CONNECTED
+} xif_mqtt_connection_states_t;
+
+typedef enum
+{
+    XIF_STATE_UNINITIALIZED = 0, /* Initialize the client -- First state in the machine */
+    XIF_STATE_LIBRARY_LOOP, /* Client is connected but idle */
+    XIF_STATE_HANDLE_ACTION_REQUESTS, /* Check requests to the event loop */
+    XIF_STATE_PAUSED, /* Client is disconnected and awaiting an external request to continue */
+    XIF_STATE_SHUTDOWN /* Shut down the connection and client */
+} xif_machine_states_t;
+
+/******************************************************************************
+*                   Xively Interface Function Declarations
+******************************************************************************/
+static int xif_init( void );
+static int xif_build_xively_topic( char* topic_name, char* dst, uint32_t dst_len );
+
+/* Application specific functions */
+static int xif_build_all_mqtt_topics( void );
+static int xif_subscribe( void );
+static int xif_start_timed_tasks( void );
+static void xif_cancel_timed_tasks( void );
+
+/* Callbacks */
+static void
+xif_on_connected( xi_context_handle_t in_context_handle, void* data, xi_state_t state );
```

Comment:
This formatting is a bit weird. See xif_led_topic_callback for fomatting that matches the rest of our codebase.

Reply:
I think each of these functions could use some comments as to what they're for, if you don't mind.  This should be a tutorial for the reader as well as a working example.

Reply:
The auto-formatter picked that format up from `.clang-format`, but it sure looks inconsistent.  Modifying it...

As to the comments: Most definitely. I'll go through documenting the entire application soon.